src/profile.c

```diff
@@ -230,8 +231,18 @@ void profile_reset_all() {
 }
 
 void profile_report_active() {
```

Comment:
The logic and purpose of this function is unclear. It appears to combine various actions, depending on state, as well as the actual reporting.